apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -908,19 +901,15 @@ get_txs_and_headers(KeepTxTypes, DropTxTypes, Account) ->
               Keep andalso not Drop
         end,
     Txs = aec_db:transactions_by_account(Account, Filter),
-    lists:map(
-        fun(SignedTx) ->
-            TxHash = aetx:hash(aetx_sign:tx(SignedTx)),
-            Header =
-                case aec_db:read_tx(TxHash) of
-                    [{mempool, _}] -> mempool;
-                    [{BlockHash, _}] ->
-                        {ok, H} = aec_chain:get_header(BlockHash),
-                        H
-                end,
-            {Header, SignedTx}
-        end,
-        Txs).
+    TxHashes = lists:usort([aetx:hash(aetx_sign:tx(SignedTx))
+                            || SignedTx <- Txs]),
+    [case aec_chain:find_transaction_in_main_chain_or_mempool(TxHash) of
```

Comment:
Shouldn't there be an outer transaction wrapping the `get_txs_and_headers/3` work?

Reply:
Agreed. It makes sense to get a proper snapshot and to avoid multiple transactions. The risk of forking in the middle of this call is relatively small, but the cost of that many transactions are probably not negligible.